mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -4610,12 +4617,23 @@ struct XdlopsGemmV2RewritePattern
     // TBD. Existing logic for fp16/bf16 may still NOT be 100% correct.
     int64_t KRepeats = 0;
     if (dataType == b.getF32Type()) {
-      KRepeats = (dataType.getWidth() / 8) / (dataType.getWidth() / 8 * k_base);
-    } else if (dataType == b.getF16Type() || dataType == b.getBF16Type()) {
+      KRepeats =
+          (dataType.template dyn_cast<FloatType>().getWidth() / 8) /
+          (dataType.template dyn_cast<FloatType>().getWidth() / 8 * k_base);
+    } else if (dataType == b.getF16Type()) {
+      VectorType argVectorType = argType.template dyn_cast<VectorType>();
+      KRepeats =
+          (dataType.template dyn_cast<FloatType>().getWidth() / 8 *
+           argVectorType.getShape()[0]) /
+          (dataType.template dyn_cast<FloatType>().getWidth() / 8 * k_base);
+    } else if (dataType == b.getIntegerType(16)) {
       VectorType argVectorType = argType.template dyn_cast<VectorType>();
-      KRepeats = (dataType.getWidth() / 8 * argVectorType.getShape()[0]) / (dataType.getWidth() / 8 * k_base);
+      KRepeats =
+          (dataType.template dyn_cast<IntegerType>().getWidth() / 8 *
```

Comment:
Factor out dataType byte-width. And it should cancel out anyway right?

Reply:
yes, already remove it